### PR TITLE
[Virtual-Wan] add command "update" for "az network vhub connection" and "az network vpn-gateway connection"

### PR DESCRIPTION
**Pull request by msyyc**
---
https://github.com/Azure/azure-cli/issues/16189
This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Virtual-Wan

File: src/virtual-wan/azext_vwan/_help.py [hunk 1]
```diff
@@ -478,6 +507,9 @@
 helps['network p2s-vpn-gateway update'] = """
     type: command
     short-summary: Update settings of a point-to-site VPN gateway.
+      - name: Update settings of a point-to-site VPN gateway with routing configuration.
```
Comment (haroldrandom) [inline src/virtual-wan/azext_vwan/_help.py:510]: Should add a key `examples` before `-name` which failed CI.

**Review by haroldrandom — CHANGES_REQUESTED** [review]
1. Please fix the complaint caused by the missing YAML indent key
2. Some of your changes might crash the old test case `test_azure_vwan_vpn_gateway_connection `

File: src/virtual-wan/azext_vwan/_help.py (hunk 1 above)
Comment (msyyc) [reply src/virtual-wan/azext_vwan/_help.py:511]: done

File: src/virtual-wan/azext_vwan/_help.py [hunk 2]
```diff
@@ -93,6 +92,21 @@
...
 """
 
+helps['network vhub connection update'] = """
+    type: command
+    short-summary: Update settings of a virtual hub connection.
+    examples:
+    - name: Add labels of a virtual hub connection.
+      text: |
+          az network vhub connection update -n MyConnection --vhub-name MyHub -g MyRG --labels Newlabel1 Newlabel2
+    - name: Add labels of a virtual hub connection.
```
Comment (kairu-ms) [inline src/virtual-wan/azext_vwan/_help.py:102]: The name of examples should be different.

File: src/virtual-wan/azext_vwan/commands.py [hunk 3]
```diff
@@ -114,6 +114,7 @@ def load_command_table(self, _):
         g.command('delete', 'delete', supports_no_wait=True, confirmation=True)
         g.show_command('show')
         g.command('list', 'list')
+        g.generic_update_command('update', custom_func_name='update_hub_vnet_connection', setter_arg_name='hub_virtual_network_connection_parameters', supports_no_wait=True)
```
Comment (kairu-ms) [inline src/virtual-wan/azext_vwan/commands.py:117]: Have your try supports_no_waits for generic_update_command?

File: src/virtual-wan/azext_vwan/commands.py [hunk 4]
```diff
@@ -150,6 +151,8 @@ def load_command_table(self, _):
         g.command('list', 'list_by_vpn_gateway')
         g.show_command('show', 'get')
         g.command('delete', 'delete')
+        g.generic_update_command('update', custom_func_name='update_vpn_gateway_connection',
```
Comment (kairu-ms) [inline src/virtual-wan/azext_vwan/commands.py:154]: same question for supports_no_waits

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**